src/main/java/frc/robot/commands/CmdDriveWithController.java

```diff
@@ -0,0 +1,57 @@
+package frc.robot.commands;
+
+import java.util.function.DoubleSupplier;
+
+import frc.robot.abstraction.SwartdogCommand;
+import frc.robot.subsystems.Drive;
+
+public class CmdDriveWithController extends SwartdogCommand
+{
+    private Drive _drive;
+
+    private DoubleSupplier _leftXInput;
+    private DoubleSupplier _leftYInput;
+    private DoubleSupplier _rightYInput;
+
+    private boolean _arcadeMode = false;
+
+    public CmdDriveWithController(Drive drive, DoubleSupplier leftXInput, DoubleSupplier leftYInput,
+            DoubleSupplier rightYInput)
+    {
+        _drive = drive;
+
+        _leftXInput = leftXInput;
+        _leftYInput = leftYInput;
+        _rightYInput = rightYInput;
+    }
+
+    @Override
+    public void execute()
+    {
+        double left = _leftYInput.getAsDouble();
+        double right = _rightYInput.getAsDouble();
+        double drive = left;
+        double rotate = _leftXInput.getAsDouble();
+
+        if (_arcadeMode)
+        {
+            _drive.arcadeDrive(drive, rotate);
+        }
+
+        else
+        {
+            _drive.tankDrive(left, right);
+        }
+    }
+
+    @Override
+    public boolean isFinished()
+    {
+        return false;
+    }
+
+    public void changeDriveMode()
+    {
+        _arcadeMode = !_arcadeMode;
+    }
```

Comment:
How are you intending this function gets called? Ultimately, we're probably going to end up tying this to a button right? But how do we bind a button to an action?

Reply:
I guess I see two ways of doing this, and I'll let you make the call on which one you want to use. We can either:
1. Put the drive mode functionality in the Subsystem and activate it with an instant command
2. Modify the command you've written to handle toggling drive mode through a `BooleanSupplier` input

I'm leaning toward the second option because if the logic is in the subsystem, the command still needs to be smart enough to figure out what joystick axes it needs to supply to the subsystem (unless you have your generic `drive` command that takes all 4 axes, but that feels like a clunky workaround to me). Since it would already need to know, it could handle the toggling as well without too much added complexity

Reply:
I was assuming we could call methods on a command using an instantcommand in order to run this function. Using a boolean supplier would still require only changing the boolean on press.

Reply:
If we're going to use an instant command (which honestly, I think this makes the most sense), I'd rather have it modify the subsystem instead of another command.

I hate to say it, but after reflecting on this, I think I like the way you did this originally 😅